lib/matplotlib/__init__.py

```diff
@@ -1352,7 +1353,10 @@ def interactive(b):
 
 def is_interactive():
     'Return true if plot mode is interactive'
-    return rcParams['interactive']
+    ret = rcParams['interactive']
+    if ret is None:
+        ret = not hasattr(main, '__file__')
+    return ret
```

Comment:
What do you mean?  This should always return a ``bool`` as ``hasattr`` returns a ``bool``.